sentinel/src/main/kotlin/com/infinum/sentinel/di/component/DomainComponent.kt

```diff
@@ -160,7 +162,9 @@ internal abstract class DomainComponent(
     @Provides
     @DomainScope
     fun sentinelWorkerFactory(): SentinelWorkerFactory =
-        SentinelWorkerFactory(collectors, notificationFactory)
+        SentinelWorkerFactory(collectors, notificationFactory).also {
+            DelegateWorker.workerFactories[CertificateCheckWorker.NAME] = it
+        }
```

Comment:
Let's avoid the put inside of a provides method because it will be executed only when someone injects SentinelWorkerFactory. That can be either very soon or sometime later - and it should be set immediately because anyone using the work manager should be able to use it without other dependencies.

Reply:
Good point 👍